packages/eslint-plugin-template/src/rules/accessibility-valid-aria.ts

```diff
@@ -30,36 +29,31 @@ export default createESLintRule<Options, MessageIds>({
   create(context) {
     const parserServices = getTemplateParserServices(context);
 
-    return parserServices.defineTemplateBodyVisitor({
-      BoundAttribute(attribute: any) {
-        validateAttribute(context, parserServices, attribute);
-      },
-      TextAttribute(attribute: any) {
-        validateAttribute(context, parserServices, attribute);
+    return {
+      'TextAttribute[name=/aria-*/], BoundAttribute[name=/aria-*/]'(
+        attribute: TmplAstTextAttribute | TmplAstBoundAttribute,
+      ) {
+        if (getAriaAttributes().has(attribute.name)) {
+          return;
+        }
+
+        const loc = parserServices.convertNodeSourceSpanToLoc(
+          attribute.sourceSpan,
+        );
+
+        context.report({
+          loc,
+          messageId: 'accessibilityValidAria',
+          data: {
+            attribute: attribute.name,
+          },
+        });
       },
-    });
+    };
   },
 });
 
-function validateAttribute(
-  context: any,
-  parserServices: any,
-  attribute: any,
-): void {
-  if (
-    !attribute.name.startsWith('aria-') ||
-    ariaAttributes.has(attribute.name)
-  ) {
-    return;
-  }
-
-  const loc = parserServices.convertNodeSourceSpanToLoc(attribute.sourceSpan);
-
-  context.report({
-    loc,
-    messageId: 'accessibilityValidAria',
-    data: {
-      attribute: attribute.name,
-    },
-  });
+let ariaAttributes: Set<string> | null = null;
+function getAriaAttributes(): Set<string> {
+  return ariaAttributes || (ariaAttributes = new Set<string>(aria.keys()));
```

Comment:
Could you explain why this function is necessary? Can't we just store it in a variable as before?

Reply:
This is a "lazy initialization", the set should not be initialized until this rule is applied. Otherwise, we will have an initialized set that we do not use.

Reply:
That makes sense!